### PR TITLE
Update React Tutorial (remove detect-metamask)

### PR DESCRIPTION
**Pull request by httpJunkie**
# Description

We want to revamp the Tutorials #1 and #2, this PR is for the update of the first of those two tutorials [Create a React dapp with local state](https://docs.metamask.io/1202-tutorial/wallet/tutorials/react-dapp-local-state/). And the intention is to keep the same idea for bnoth tutorials with updates to the code and explanation using EIP-6963 instead of `window.ethereum` and `@metamask/detect-provider`. 

I may add some links out to existing pages that speak on EIP-6963 from the tutorial rather than covering the topic as we already have docs "Wallet Interoperability" and "Detecting MetaMask" that we could create cross links to.

## Issue(s) fixed

Tutorial instruction and code have been update, I still need to link to an updated GitHub Repo so that when this goes live the prior repo is not affected until then.

Fixes #

## Preview

https://github.com/MetaMask/metamask-docs/assets/4899480/0790319f-03dd-4a8c-b18f-0059cddc2ccf

## Checklist

Complete this checklist before merging your PR:

- [x] If this PR contains a major change to the documentation content, I have added an entry to the top of the ["What's new?"](https://github.com/MetaMask/metamask-docs/blob/main/docs/whats-new.md) page.
- [ ] The proposed changes have been reviewed and approved by a member of the documentation team.

### CODE REVIEW

**Review by alexandratran — COMMENTED** [review]
Thanks for this update! A few initial comments:

1. I think we can get rid of step **2. Discover Wallet Providers (Browser Extensions)**. The beginner tutorial (#1276) should cover this.
2. Can we split the current step **3. Detect Multiple Wallet Providers** into multiple steps or multiple sub-steps so it's not so long? Maybe setting up the environment variables, creating the hooks, and adding the buttons and CSS. If this makes sense to you I can go ahead and add those edits.
3. Can you add screenshots to this page demonstrating what the user might see? We need to update the screenshot at the end of step 1, and add another at the end of the tutorial.

**Comment by httpJunkie** [comment]
Tutorial #1 (React with local state) above is the final draft from @httpJunkie (Eric Bishard).
We need to approve and wait and send the React with Global State tutorial out at the same time together.

But we can still get this review approved and out of the way.

File: wallet/assets/react-tutorial-01-final.png
Comment (alexandratran) [inline wallet/assets/react-tutorial-01-final.png:1]: The two images provided are the same. Should this instead be a screenshot of the final dapp which displays an address for the connected wallet?

**Review by alexandratran — COMMENTED** [review]
Edited the content for clarity and to create more digestible steps. One question about the screenshots provided.

File: wallet/assets/react-tutorial-01-final.png
Comment (httpJunkie) [reply wallet/assets/react-tutorial-01-final.png:1]: I may have used the same image twice, was not the intention, let me look at it.

Comment (httpJunkie) [reply wallet/assets/react-tutorial-01-final.png:1]: Thats fixed, also I see how you did the CSS and it's funny. That's how I had originally had the article formatted and then I changed something and got lazy. lol It's perfect now. Thank you @alexandratran

**Review by alexandratran — APPROVED** [review]
LGTM!

**Pull request merged**